eox_nelp/pearson_vue/pipeline.py

```diff
@@ -16,10 +18,46 @@
 from eox_nelp.edxapp_wrapper.student import anonymous_id_for_user
 from eox_nelp.pearson_vue.constants import PAYLOAD_CDD, PAYLOAD_EAD, PAYLOAD_PING_DATABASE
 from eox_nelp.pearson_vue.utils import update_xml_with_dict
+from eox_nelp.signals.utils import get_completed_and_graded
+
+logger = logging.getLogger(__name__)
 
 User = get_user_model()
 
 
+def check_completion_metadata(user_id, course_id, **kwargs):
+    """Pipeline that check the case of completion cases on the pipeline execution. Also this pipe
+    has 3 behaviours depending the case:
+        - is_passing is true means the course is graded(passed) and dont needs this pipe validation.
+          The pipeline continues without changes.
+        - is_complete=True and is_graded=False pipeline should continue with uploaded metadata
+          (completed courses and not graded).
+        - Otherwise the pipeline will stop, for grading-courses the COURSE_GRADE_NOW_PASSED signal would act.
+
+    Args:
+        user_id (int): The ID of the user whose data is to be retrieved.
+        course_id (str): course_id to check completion or graded.
+        **kwargs: Additional keyword arguments.
+
+    Returns:
+        dict: Pipeline dict
+    """
+    if kwargs.get("is_passing"):
```

Comment:
@andrey-canon I was thinking in this and I could skip the get_completed_and_graded call for grading course without the suggestion. That method uses modulestore, so it skips a call in performance.
Thinking more the better performance approach is using only the receiver of COURSE_GRADE_NOW_PASSED in order, to skip a lot of modulestore call thinking the completion of each unit, of each user, of each course =O xD.
But we could still have both approaches